modules/rostests/apitests/comctl32/imagelist.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         GPL - See COPYING in the top level directory
+ * PURPOSE:         Test for imagelist
+ * PROGRAMMERS:     Whindmar Saksit
+ */
+
+#include "wine/test.h"
+#include <stdio.h>
+#include <windows.h>
+#include <commctrl.h>
+#include <comctl32_undoc.h>
+
+#define WinVerMajor() LOBYTE(GetVersion())
+
+#define ILC_CLRMASK (0xff & ~ILC_MASK)
+#define IL_IMGSIZE 16
+
+
+static BOOL IL_IsValid(HIMAGELIST himl)
+{
+    int w = -42, h;
+    if (!himl || IsBadReadPtr(himl, sizeof(void*)))
+        return FALSE;
+    return ImageList_GetIconSize(himl, &w, &h) && w != -42;
+}
+
+static HRESULT IL_Destroy(HIMAGELIST himl)
+{
+    if (himl && !IL_IsValid(himl))
+        return E_INVALIDARG;
+    return ImageList_Destroy(himl) ? S_OK : S_FALSE;
+}
+
+static inline HIMAGELIST IL_Create(UINT flags)
+{
+    return ImageList_Create(IL_IMGSIZE, IL_IMGSIZE, flags, 1, 0);
+}
+
+static BOOL IL_AddImagesForTest(HIMAGELIST himl)
+{
+    int idx = -1;
+    HINSTANCE hInst = LoadLibraryW(L"USER32");
```

Comment:
```suggestion
    HINSTANCE hInst = GetModuleHandleW(L"USER32");
```
(we know user32 is already loaded).
Also, the LoadLibrary call would require a FreeLibrary, that you didn't call and this would leak a reference.

Reply:
We know user32 is loaded (and probably statically linked) so there will be no leak. But OK...